force-app/main/default/lwc/marketingCloudCopadoDataTable/marketingCloudCopadoDataTable.js

```diff
@@ -308,9 +308,11 @@ export default class MarketingCloudCopadoDataTable extends LightningElement {
                     "/events/copado/v1/step-monitor/"
                 )
             ) {
-                // show progress on screen
-                const stepStatus = JSON.parse(response.data.payload.copado__Payload__c);
-                this.progressStatus = stepStatus.data.progressStatus || this.progressStatus;
+                if (this.isJSON(response.data.payload.copado__Payload__c)) {
+                    // show progress on screen
+                    const stepStatus = JSON.parse(response.data.payload.copado__Payload__c);
+                    this.progressStatus = stepStatus.data.progressStatus || this.progressStatus;
+                }
             }
```

Comment:
instead i would go with the approach that is described here: https://stackoverflow.com/a/3710226/818732

```suggestion
                try {
                    // show progress on screen
                    const stepStatus = JSON.parse(response.data.payload.copado__Payload__c);
                    this.progressStatus = stepStatus.data.progressStatus || this.progressStatus;
                } catch {}
            }
```

this would mean we don't need some custom function to determine if the json is valid

Reply:
response data field appended with <p>response data </p>. so put validation

Reply:
dude. we went over this in our call. all I wanted you to do here is click on "Cmmit suggestion" followed by a click on the then visible "Commit changes" button. why didn't you? I literally showed you that during our call.

![image](https://user-images.githubusercontent.com/1917227/195384508-5c693a4c-f290-48bb-bef9-f369330249e6.png)